codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/TestHelpers.kt

```diff
@@ -94,17 +93,24 @@ internal fun testCodegenContext(
     codegenTarget,
 )
 
+interface BuilderGenerator {
+    fun render(writer: RustWriter)
+    fun renderConvenienceMethod(implBlock: RustWriter)
+}
```

Comment:
I'm not a fan of this since it makes the builders inherit from unit test code, but it seems like this might go away due to #2342? Perhaps we merge that PR first and then delete this `BuilderGenerator` interface.